deber/main.tex

```diff
@@ -0,0 +1,281 @@
+%========= info ===================
+% Esta es una plantilla para reportes de laboratorio
+% Autor: Alex Soto
+% Licencia: 
+% ================================
+
+\documentclass[12pt]{article} % Declarando clase de documento
+%======= pre-ambulo ==========%
+\usepackage[utf8]{inputenc}
+% \usepackage[backend=biber, style=ieee, sorting=ynt]{biblatex}
+\usepackage[backend=biber, style=ieee]{biblatex}
+\addbibresource{referencias.bib}
+
+%===== Definiendo Idioma Español =======
+\usepackage[spanish]{babel}
+\selectlanguage{spanish}
+%===== Control de Margenes ======
+\usepackage[a4paper,tmargin=3cm,bmargin=2.5cm,lmargin=3cm, rmargin=2.5cm, bindingoffset=6mm]{geometry}
+%======================================================================================================
+%======== libreira para poner color en un cuadro
+\usepackage{framed}
+\usepackage{xcolor}
+\definecolor{shadecolor}{rgb}{255,255,255}
+%======================================
+%==========Libreria para Justificar===
+\usepackage{ragged2e} 
+% ======= librery graphicx=========
+\usepackage{graphicx}
+% =================================
+% ===== librerias matematicas ==========
+\usepackage{amsmath, amssymb, amsfonts}
+\usepackage{amsthm}
+\usepackage{mathrsfs}
+%========= Enlaces
+\usepackage{hyperref}
+
+
+
+\begin{document}
+
+\title{Teorema maestro}
+\hrulefill
+\justify En el análisis de algoritmos, el \textbf{teorema maestro} proporciona una solución sencilla en términos asintóticos (usando una Cota superior asintótica) para ecuaciones de recurrencia que ocurren en muchos algoritmos recursivos tales como en el Algoritmo divide y vencerás. Fue popularizado por el libro Introducción a los algoritmos por Cormen, Leiserson, Rivest, y Stein, en el cual fue tanto introducido como probado formalmente. No todas las ecuaciones de recurrencia pueden ser resueltas con el uso del teorema maestro.
+
+\tableofcontents
+
+%\section*{Introduccion}
+
+\justify Considere un problema que puede ser resuelto a través de un algoritmo recursivo como el siguiente:
+
+\begin{shaded*}
+    
+\hspace{-0.75cm}\textbf{procedimiento} T( n : tamaño del problema ) \textbf{definición:}\newline
+    \hspace{1cm}\textbf{if} $n < 1$ \textbf{then exit}
+        
+\begin{verbatim}
```

Comment:
Considera utilizar algun paquete para escribir algoritmos como el siguiente código por ejemplo
\usepackage[ruled,vlined]{algorithm2e}
\begin{algorithm}[htbp]
\SetAlgoLined
\SetKw{exit}{exit}
\SetKwProg{Fn}{Procedimiento}{:}{end}
\Fn{T(n: tamañoProblema)}{
        \If(){
            $n<1$}{exit\;}
        \Else(){
            \Repeat{$a$ veces}{$T\left(n/b\right)$\;}
        }

    }

 \caption{How to write algorithms}
\end{algorithm}